splicemachine/spliceengine
# DB-12107 Close result set when exception is thrown

## Description

When an exception is thrown in `GenericPreparedStatement.executeStmt`, a result set may be left open forever. 
This PR closes the result set if we know it will not be used anymore.

## How to test

See the added test in `SpliceUDTIT` for how to test this. We defined a custom java function that throws an NPE. Before this change, we would not be able to drop the tables. After this change, dropping tables works properly.
